boto/utils.py

```diff
@@ -864,7 +864,7 @@ def get_utf8_value(value):
         return value
 
     if not isinstance(value, six.string_types):
-        value = six.text_type(value)
+        value = six.text_type(value).encode('utf-8')
```

Comment:
I don't think this change is necessary.  If we cast `value` to type `six.text_type`, then the next if statement should evaluate to True and do the encoding for us.